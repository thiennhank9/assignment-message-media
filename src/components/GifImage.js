export default function GifImage(props) {
  const { src, alt, onClick, ...restProps } = props;

  return (
    <div className="gif-image-container">
      <img
        className="gif-image"
        src={src}
        onClick={onClick}
        alt={alt}
        {...restProps}
      />
    </div>
  );
}
