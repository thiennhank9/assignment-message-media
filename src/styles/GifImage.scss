.gif-image-container {
  width: 100%;
  padding-bottom: 100%;
  position: relative;
}

.gif-image {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
