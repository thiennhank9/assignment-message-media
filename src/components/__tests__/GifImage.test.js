import renderer from "react-test-renderer";
import GifImage from "../GifImage";

test("GifImage - render correctly", () => {
  const tree = renderer.create(<GifImage />).toJSON();
  expect(tree).toMatchSnapshot();
});
