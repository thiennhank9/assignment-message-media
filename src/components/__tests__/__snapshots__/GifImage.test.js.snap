// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifImage - render correctly 1`] = `
<div
  className="gif-image-container"
>
  <img
    className="gif-image"
  />
</div>
`;
